File: src/test/java/searchinsert/SolutionTest.java
package searchinsert;

import java.util.stream.Stream;
import org.assertj.core.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

class SolutionTest {

    private final Solution solution = new Solution();

    @ParameterizedTest
    @MethodSource("testOneData")
    public void testOne(int[] nums, int target, int expectedOutput) {
        int output = solution.searchInsert(nums, target);

        Assertions.assertThat(output).isEqualTo(expectedOutput);
    }

    private static Stream<Arguments> testOneData() {
        return Stream.of(
//            Arguments.of(new int[]{1, 3, 5, 6}, 5, 2),
//            Arguments.of(new int[]{1, 3, 5, 6}, 2, 1),
//            Arguments.of(new int[]{1, 3, 5, 6}, 7, 4),
//            Arguments.of(new int[]{1, 3}, 0, 0),
            Arguments.of(new int[]{1, 2, 4, 6, 7}, 3, 2)
        );
    }
}

File: src/test/kotlin/dev/sanket/minsubarraysize/SolutionTest.kt
package dev.sanket.minsubarraysize

import assertk.assertThat
import assertk.assertions.isEqualTo
import dev.sanket.dev.sanket.minsubarraysize.Solution
import java.util.stream.Stream
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

class SolutionTest {

    private val solution = Solution()

    @ParameterizedTest
    @MethodSource("testData")
    fun shouldPass(nums: IntArray, target: Int, size: Int) {
        val length = solution.minSubArrayLen(target, nums)
        assertThat(length).isEqualTo(size)
    }

    companion object {

        @JvmStatic
        fun testData(): Stream<Arguments> = Stream.of(
            Arguments.of(intArrayOf(2, 3, 1, 2, 4, 3), 7, 2),
            Arguments.of(intArrayOf(1, 4, 4), 4, 1),
            Arguments.of(intArrayOf(1, 1, 1, 1, 1, 1, 1, 1), 11, 0),
        )
    }
}